scripts/update-charon-pin.sh

```diff
@@ -10,11 +10,11 @@ if [ -L charon ]; then
     nix flake lock --override-input charon "github:aeneasverif/charon/$COMMIT"
 else
     echo 'Pinning the latest commit from Charon `main`'
-    nix flake lock --update-input charon
+    nix flake update charon
 fi
 # In both cases we also update the rust overlay to make sure we can get the
 # latest rust nightlies.
-nix flake lock --update-input charon/rust-overlay
+nix flake update charon/rust-overlay
```

Comment:
I will leave the review of this to Guillaume :)

Reply:
@Nadrieril I think it may not work on Nix 2.18, but it broke on more modern Nix, so not sure what to do. I guess command testing makes more sense here.

Reply:
I've used that before on 2.18, why may this not work?

Reply:
I'm pretty sure there's a Nix version that doesn't support `nix flake update XYZ`, I thought it was 2.18 but I guess I'm wrong.

Reply:
ah, it does not look like it does what I want indeed. `nix flake update rust-overlay` on charon fails with `cannot find flake 'flake:rust-overlay' in the flake registries` which is not what I want

Reply:
I can drop this part of the PR, tbh. This was more chores that I picked up as I went trying to update my repo.

Reply:
Dropped!